### PR TITLE
Fix EMS actuators for Zone Temperature Control

### PR DESCRIPTION
**Pull request by lgu1234**
Pull request overview
---------------------
EMS actuators for zone temperature control were not working as intended. EMS actuators for zone temperature control now update the thermostat temperature setting after thermostat temperature schedules are initialized. This change corrects issue #5870.

### Work Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [x] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [x] At least one of the following appropriate labels must be added to this PR to be consumed into the changelog:
   - Defect: This pull request repairs a github defect issue.  The github issue should be referenced in the PR description
   - Refactoring: This pull request includes code changes that don't change the functionality of the program, just perform refactoring
   - NewFeature: This pull request includes code to add a new feature to EnergyPlus
   - Performance: This pull request includes code changes that are directed at improving the runtime performance of EnergyPlus
   - DoNoPublish: This pull request includes changes that shouldn't be included in the changelog

### Review Checklist
This will not be exhaustively relevant to every PR.
 - [x] Code style (parentheses padding, variable names)
 - [x] Functional code review (it has to work!)
 - [x] If defect, results of running current develop vs this branch should exhibit the fix
 - [x] CI status: all green or justified
 - [x] Performance: CI Linux results include performance check -- verify this
 - [x] Unit Test(s)
 - C++ checks:
   - [ ] Argument types
   - [ ] If any virtual classes, ensure virtual destructor included, other things
 - IDD changes:
   - [ ] Verify naming conventions and styles, memos and notes and defaults
   - [ ] Open windows IDF Editor with modified IDD to check for errors
   - [ ] If transition, add rules to spreadsheet
   - [ ] If transition, add transition source
   - [ ] If transition, update idfs
 - [ ] If new idf included, locally check the err file and other outputs
 - [ ] Documentation changes in place
 - [ ] Changed docs build successfully
 - [ ] ExpandObjects changes?
 - [ ] If output changes, including tabular output structure, add to output rules file for interfaces

### CODE REVIEW

**Comment by lgu1234** [comment]
@EnergyArchmage I created a new function: OverrideAirSetPointsforEMSCntrl and moved all EMS code in the Init function to this new function. This function is called at the end of CalcZoneAirTempSetPoints, by assuming all EMS revision should be final changes. I did not place my changes along side of AdjustAirSetPointsforOpTempCntrl. Here is my concern:

1) If EMS is placed along side of AdjustAirSetPointsforOpTempCntrl, it will call every temperature control type. In order to be compact, I make a call only once.
2) Keep original integrity and easy to read.

By the way, a test file was uploaded. No differences were found.

I check-in the code only for review and comments. If OK, I will make a unit test.

**Comment by EnergyArchmage** [comment]
Thanks @lgu1234 , this looks good to me.

**Comment by lgu1234** [comment]
@EnergyArchmage Thanks for review. I am going to check-in a unit test.

**Comment by nrel-bot-2** [comment]
@lgu1234 @lgentile it has been 16 days since this pull request was last updated.

**Comment by nrel-bot-3** [comment]
@lgu1234 @lgentile it has been 15 days since this pull request was last updated.

**Comment by nrel-bot-3** [comment]
@lgu1234 @lgentile it has been 14 days since this pull request was last updated.

File: src/EnergyPlus/ZoneTempPredictorCorrector.cc [hunk 1]
```diff
@@ -6818,6 +6731,148 @@ namespace ZoneTempPredictorCorrector {
...
+		// SUBROUTINE ARGUMENT DEFINITIONS:
+
+		// SUBROUTINE PARAMETER DEFINITIONS:
+		// na
+
+		// INTERFACE BLOCK SPECIFICATIONS:
+		// na
+
+		// DERIVED TYPE DEFINITIONS:
+		// na
```
Comment (rraustad) [inline src/EnergyPlus/ZoneTempPredictorCorrector.cc]: @lgu1234 we no longer include comment lines if they are not needed. Delete these lines that include "// na"

Comment (lgu1234) [reply src/EnergyPlus/ZoneTempPredictorCorrector.cc]: @rraustad Done. I am going to upload the file.

**Comment by rraustad** [comment]
This looks very good. The defect file has cooling set point temperature of 24C with 30C setback.

  Schedule:Compact,
    Cooling Setpoints,       !- Name
    Temperature,             !- Schedule Type Limits Name
    Through: 12/31,          !- Field 1
    For: AllDays,            !- Field 2
    Until:  7:00,30.0,       !- Field 3
    Until: 17:00,24.0,       !- Field 5
    Until: 24:00,30.0;       !- Field 7

The EMS program overrides that setting with 26C

EnergyManagementSystem:Actuator,
EMS_Cooling_setpoint, ! Name
EAST ZONE, ! Component Name
Zone Temperature Control, ! Component Type
Cooling Setpoint; ! Control Variable

EnergyManagementSystem:Program ,
EMS_Cooling_setpoint_Program, ! Name
SET EMS_Cooling_setpoint = 26;

EnergyManagementSystem:ProgramCallingManager,
EMS_Call,
BeginTimestepBeforePredictor,
EMS_Cooling_setpoint_Program;

The simulation results show the correct Tstat temperature set by the EMS.

**Comment by rraustad** [comment]
Merged and deleted branch.

**Pull request merged**